tests/test_core.py

```diff
@@ -88,6 +88,21 @@ def decorated(self):
   return decorated
 
 
+# Tests exception handling in emscripten exception handling mode, and if
+# possible, new wasm EH mode.
+def with_both_exception_handling(f):
+  def decorated(self):
+    self.set_setting('DISABLE_EXCEPTION_CATCHING', 0)
+    f(self, None)
+    self.set_setting('DISABLE_EXCEPTION_CATCHING', 1)
```

Comment:
Why set disable to 1 here? then it stays disabled when we run the wasm EH test below.

(Or when `EXCEPTION_HANDLING` is set, do we ignore the value of `DISABLE_EXCEPTION_CATCHING`? that seems risky for code in our JS library I think.)

Reply:
I was using `DISABLE_EXCEPTION_CATCHING` as on/off switch for Emscripten EH and `EXCEPTION_HANDLING` as that for the wasm EH. Do you think this can be problematic? You mean, we should think wasm EH is used only when both `DISABLE_EXCEPTION_CATCHING == 0` and `EXCEPTION_HANDLING == 1` are satisfied, and if only `DISABLE_EXCEPTION_CATCHING == 0` is satisfied if automatically assume Emscripten EH? Then I need to fix the driver part. Before I do that please confirm this is what you meant.

Reply:
Thinking on this more, on the one hand, I think it makes sense that `DISABLE_EXCEPTION_CATCHING` means no exceptions, period. If that setting is off, then if `WASM_EXCEPTIONS` is on we use wasm exceptions, otherwise we use the emulated old approach.

But on the other hand, if we want to some day remove the old emulated exceptions, perhaps it would be better to not need that old flag.

I am leaning towards the way you have it currently, actually. Maybe just documenting it more in settings.js might avoid confusion?

Reply:
Added more comments on settings.js.